src/Orleans.Runtime.Abstractions/Facet/Persistent/IPersistentState.cs

```diff
@@ -0,0 +1,9 @@
+using Orleans.Core;
+
+namespace Orleans.Runtime
+{
+    public interface IPersistentState<TState> : IStorage<TState>
+        where TState : new()
```

Comment:
Minor: we could drop the `new()` constraint on `TState` by using `ActivatorUtilities`. I don't believe there are any major downsides. Maybe there are reasons that a user would want to inject services into their state, maybe not.